### ISSUE TITLE
Pink Color Issue in Full Screen (for macOS Sonoma version 14.5)

### ISSUE THREAD

**Issue by slook** [nicotine-plus/nicotine-plus#3066]
### Discussed in https://github.com/nicotine-plus/nicotine-plus/discussions/3065

<div type='discussions-op-text'>

<sup>Originally posted by **apartmandairesi** July 21, 2024</sup>
Hello.

I am using **macOS Sonoma version 14.5**. I have tried to install Nicotine+-3.3.4. However when I've opened app, I've saw an error which Is **pink color** in full screen. Do you have any idea about this issue? Thank you.</div>

**Comment by slook** [comment]
> ![image](https://github.com/user-attachments/assets/74ac804c-b002-40b3-aebc-b9ed1305549d)

`Gdk-CRITICAL: _gdk_macos_surface_update_fullscreen_state: assertion 'GDK_IS_MACOS_SURFACE (self)' failed`

**Comment by slook** [comment]
@ArrBianca @cattz Are you also encountering this crash? If so, please can you tell us information about your hardware what kind of Mac do you have?

**Comment by LordBabass** [comment]
Hello,
(i am a new Mac user) I also meet this bug (pink flashing screen). 
It is replicable: I think it has to deal with UI scaling to monitor's resolution.

I think so because I use "Better Display" (so I use scaling that is not available natively in MacOS)... I have a 32 inch 4k monitor and I want more space than 1440p but not not too much or else I think everything is too small !
<img width="296" alt="SCR-20241004-oqmt" src="https://github.com/user-attachments/assets/9448dcf3-1925-4cb2-9d0c-9037787deb30">

You can launch Nicotine+, then change UI (fine tweaking % by %) and you will see that some resolutions do flash pink, while others don't suffer it.
Nicotine+ doesn't crash when it happens... only the UI flickering quickly to pink

**Comment by slook** [comment]
Then this bug either belongs to the Better Display app or GTK. Do you know if the same problem occurs with other Gtk/GNOME based apps such as [Gedit](https://formulae.brew.sh/formula/gedit) for example?

**Comment by mathiascode** [comment]
The development build also switched to GTK's new GL renderer today. It could be worth checking if that solved the issue: https://nicotine-plus.org/doc/TESTING.html#macos

**Comment by LordBabass** [comment]
Hello, I just tried the new 3.3.6 DEV version, but the problem is still present under certain resolutions...

I imagine that Better Display is no culprit, since it only changes scaling % resolutions.
Also, I never have seen this behavior with any other software than Nicotine+

I don't know Gedit or Gaphor

I'm sorry I can't help any more, I am no developer... (I created a GitHub account to contribute to this topic with my observations...)

**Comment by slook** [comment]
@LordBabass Please try the latest build of 3.3.7rc2 to verify it at least starts up, and if the display issue is the same or different?
https://nicotine-plus.org/doc/TESTING.html#macos

Thank you for your time helping with testing, it's very useful.

**Comment by LordBabass** [comment]
Hi,

I have tried 3.3.7RC2 version, the display still has the strange behavior under certain resolutions (shaking very quickly / tinting pink)

Best regards

**Comment by mathiascode** [comment]
> I am experiencing the exact same problem on my Macbook Pro M3 with latest MacOS installed.

Are you using something like Better Display to set a custom screen resolution (see https://github.com/nicotine-plus/nicotine-plus/issues/3066#issuecomment-2393871171)?

**Comment by eazy4me** [comment]
> > I am experiencing the exact same problem on my Macbook Pro M3 with latest MacOS installed.
> 
> Are you using something like Better Display to set a custom screen resolution (see [#3066 (comment)](https://github.com/nicotine-plus/nicotine-plus/issues/3066#issuecomment-2393871171))?

Yes I am using a 40 inch monitor. So that might be the problem I assume

**Comment by mathiascode** [comment]
> Yes I am using a 40 inch monitor. So that might be the problem I assume

Does the issue happen with this build (download links at the bottom)? https://github.com/mathiascode/nicotine-plus/actions/runs/12619743654

**Comment by codyvme** [comment]
> > Yes I am using a 40 inch monitor. So that might be the problem I assume
> 
> Does the issue happen with this build (download links at the bottom)? https://github.com/mathiascode/nicotine-plus/actions/runs/12619743654

Another BetterDisplay user here checking in. Unfortunately I'm still getting the pink screen bug with that build too.

**Comment by mathiascode** [comment]
Just to clarify a few things:
- What exactly do you mean by full screen? Is the macOS top bar and dock visible?
- Is only the Nicotine+ window pink, or everything on screen? Are you still able to access other applications?
- Does the issue happen with an unmaximized window at all?

**Comment by codyvme** [comment]
I recorded a [quick video](https://youtu.be/CxAr2h_ZoqA) showing you the quirks of this bug. **BE WARNED**, there is rapidly flashing parts in this video if whoever is reading this is sensitive to such things.

- You still have access to the macOS menu bar and can launch the Nicotine+ settings but that's about it
- Trying to switch to another program or even close Nictotine+ gets really laggy
- The header bar setting in "User Interface" doesn't seem to make a difference
- Resizing the window really gets weird in the second half of the video

**Comment by mathiascode** [comment]
Thank you for recording the video. What monitor are you using, and what's the current resolution set to?

**Comment by codyvme** [comment]
It's a 4k monitor scaled to 2880x1620 (~150%)

**Comment by mathiascode** [comment]
Are you able to reproduce the issue in another GTK 4 application, like [Gaphor](https://gaphor.org/)?

**Comment by codyvme** [comment]
It does not happen in Gaphor. The Gaphor window doesn't open fully
maximized on first start though like Nicotine+ does. I'm wondering if
that's part of it?

**Comment by mathiascode** [comment]
> It does not happen in Gaphor. The Gaphor window doesn't open fully maximized on first start though like Nicotine+ does. I'm wondering if that's part of it?

I think the main Gaphor window should remember the maximized state across restarts. Any difference if you maximize it?

**Comment by codyvme** [comment]
I forgot to try that yesterday when I was testing Gaphor. If I do try and maximize the Gaphor window, it does some weird flickering and a couple seconds later gives me the pink screen bug. I guess this is a problem with GTK 4 applications at the moment.

**Comment by mathiascode** [comment]
Tracked upstream in https://gitlab.gnome.org/GNOME/gtk/-/issues/7244
